pkg/eosclient/eosgrpc/eoshttp.go

```diff
@@ -390,9 +390,6 @@ func (c *EOSHTTPClient) PUTFile(ctx context.Context, remoteuser string, auth eos
 		log.Debug().Str("func", "PUTFile").Msg("sending req")
 
 		resp, err := c.doReq(req, remoteuser)
```

Comment:
You should check the error `err` here. If the error is nil, then the `resp.Body` is guaranteed by the stdlib to be non-nil, and you can defer the close of the body without checking for it to be non-nil. If you do this, you can remove all the manual body close below.